src/lib.rs

```diff
@@ -107,6 +116,22 @@ impl hal::blocking::delay::DelayMs<u64> for Delay {
     }
 }
 
+
+#[cfg(all(feature = "gpio_sysfs", feature = "gpio_cdev"))]
+/// Re-export of `sysfs_pin::SysfsPin` when both pin types are enabled
+/// This exists to maintain backwards compatibility with existing users
+pub type Pin = sysfs_pin::SysfsPin;
+
+#[cfg(all(feature = "gpio_sysfs", not(feature = "gpio_cdev")))]
+/// Re-export of `sysfs_pin::SysfsPin` pin type when `gpio_sysfs` feature is selected
+pub type Pin = sysfs_pin::SysfsPin;
+
+#[cfg(all(feature = "gpio_cdev", not(feature = "gpio_sysfs")))]
+/// Re-export of `cdev_pin::CdevPin` pin type when `gpio_cdev` feature is selected
+pub type Pin = cdev_pin::CdevPin;
```

Comment:
My gut feeling is that it could make sense to re-export `SysfsPin` as `Pin` if the gpio_sysfs feature is enabled.  I think I would prefer to not export any symbol with the name `Pin` that ever points at a different type that is cdev.  I think that's probably just going to cause confusion without any real benefit.

Reply:
That's probably fair, I put this in because a) there was a duplicated Pin impl and b) it makes the change non-breaking. 

If we're going to break it, does it make sense to have the re-export at all? We could just export whatever is enabled and indicate a preference in the comments?

Reply:
I guess that would be my preference overall -- With a semver bump I think the move to be explicit about the subsystem being used to talk to the GPIO makes sense.  Any breakage with the semver major version is not something I think will be a big deal for users if they are sticking with sysfs.

Reply:
sweet as, updated